lldb/include/lldb/Core/Progress.h

```diff
@@ -55,6 +55,11 @@ namespace lldb_private {
 
 class Progress {
 public:
+  /// Enum that indicates the type of progress report
+  enum class ProgressReportType {
+    eAggregateProgressReport,
+    eNonAggregateProgressReport
```

Comment:
I wonder if it might be a little easier to intuit if the enums were eAggregateProgressReport and eIndividualProgressReport?  Then someone might look at this and say "oh I have a single thing I'm providing progress on, I probably want eIndividualProgressReport".  I don't feel strongly that I'm correct here, just throwing it out.

Reply:
Maybe eSolitaryProgressReport

Reply:
eUmbrellaProgressReport may work here since that's what's happening when we use a single progress report that's being incrementally updated.

Reply:
> eUmbrellaProgressReport may work here since that's what's happening when we use a single progress report that's being incrementally updated.

What is happening multiple times a session? We only index the DWARF one time where we have N DWARF units that we index, but that doesn't happen at all for dSYM files. Symbol tables are parsed only once, but we do parse the symbol tables for each library. What is happening multiple times and how are we going to be able to group these?

Would it be better to use progress category as a string instead of a bool? I will make some inline comments to clarify this.